emperor/core.py

```diff
@@ -302,39 +303,43 @@ def _repr_html_(self):
 
         return display(HTML(str(self)))
 
-    def _validate_metadata(self, metadata, matrix, ignore_missing_samples):
+    def _validate_metadata(self, metadata, matrix, ignore_missing_samples,
+                           kind='sample'):
+
+        if kind not in {'sample', 'feature'}:
+            raise ValueError('Unsupported "kind" value %s' % kind)
 
         # metadata is optional for biplots, so we just create an empty table
         if metadata is None:
             metadata = pd.DataFrame(index=pd.Index(matrix.index, name='id'))
-            metadata['all'] = 'All objects'
+            metadata['all'] = 'All elements'
             return metadata
 
-        ordination_samples = set(matrix.index)
-        difference = ordination_samples - set(metadata.index)
+        ordination_elements = set(matrix.index)
+        difference = ordination_elements - set(metadata.index)
 
-        if difference == ordination_samples:
-            raise ValueError('None of the sample identifiers match between the'
+        if difference == ordination_elements:
+            raise ValueError('None of the %s identifiers match between the'
                              ' metadata and the coordinates. Verify that you '
                              'are using metadata and coordinates corresponding'
-                             ' to the same dataset.')
+                             ' to the same dataset.' % kind)
 
         if difference and not ignore_missing_samples:
-            raise KeyError("There are samples not included in the mapping "
+            elements = ', '.join(sorted([str(i) for i in difference]))
```

Comment:
Can this be a truncated list? If thousands of features are missing, the output will be substantial especially if they're sOTUs

Reply:
I think it's ok since QIIME2 would hide the list of features by default and only if the user uses `--verbose` the full text will be displayed. What do you think?

Reply:
Does it...? I ran q2-emperor from 2019.7 recently with malformed metadata and got a buffer full of output on the terminal :/

Reply:
Got it, just addressed this 👍